### PR TITLE
MM-20749: restrict post menu nesting

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Submenus in the RHS may exceed the width of the RHS, and are currently truncated. While this will happen with any long submenu text, it's especially painful when multiple levels of nesting are allowed.

For now, disable more than one submenu nesting level.

#### Ticket Link
Fixes: https://mattermost.atlassian.net/browse/MM-20749

### CODE REVIEW

File: plugins/registry.js [hunk 1]
```diff
@@ -271,6 +273,10 @@ export default class PluginRegistry {
                 },
             });
             return function registerSubMenuItem(t, a, f) {
+                if (parentMenuId) {
+                    throw new Error('Submenus are currently limited to a single level.'); // eslint-disable-line no-console
```
Comment (marianunez) [inline plugins/registry.js]: Do we need the `eslint-disable-line no-console` here?

Comment (lieut-data) [reply plugins/registry.js]: Oops, nope!
```suggestion
                    throw new Error('Submenus are currently limited to a single level.');
```

**Comment by prapti** [comment]
@lieut-data The change here is tiny but would any unit test coverage help?

**Pull request merged**